gcsio/src/main/java/com/google/cloud/hadoop/gcsio/StorageStubProvider.java

```diff
@@ -142,43 +130,24 @@ public void shutdown() {
     }
   }
 
-  public static StorageStubProvider newInstance(
-      GoogleCloudStorageOptions options,
-      ExecutorService backgroundTasksThreadPool,
-      Credential credential) {
-    boolean defaultServiceAccount =
-        credential != null
-            && java.util.Objects.equals(
-                credential.getTokenServerEncodedUrl(), ComputeCredential.TOKEN_SERVER_ENCODED_URL);
-    GrpcDecorator grpcDecorator =
-        getGrpcDecorator(
-            options,
-            defaultServiceAccount,
-            new CloudPathGrpcDecorator(new CredentialAdapter(credential)));
-    return new StorageStubProvider(options, backgroundTasksThreadPool, grpcDecorator);
-  }
-
   public static StorageStubProvider newInstance(
       GoogleCloudStorageOptions options,
       ExecutorService backgroundTasksThreadPool,
       Credentials credentials) {
-    boolean defaultServiceAccount = credentials instanceof ComputeEngineCredentials;
     return new StorageStubProvider(
-        options,
-        backgroundTasksThreadPool,
-        getGrpcDecorator(options, defaultServiceAccount, new CloudPathGrpcDecorator(credentials)));
+        options, backgroundTasksThreadPool, getGrpcDecorator(options, credentials));
   }
 
   private static GrpcDecorator getGrpcDecorator(
-      GoogleCloudStorageOptions options,
-      boolean defaultServiceAccount,
-      CloudPathGrpcDecorator credential) {
-    if (options.isTrafficDirectorEnabled() && defaultServiceAccount) {
-      return new TrafficDirectorGrpcDecorator();
-    }
-    if (options.isDirectPathPreferred() && defaultServiceAccount) {
-      return new DirectPathGrpcDecorator();
+      GoogleCloudStorageOptions options, Credentials credentials) {
+    if (credentials instanceof ComputeEngineCredentials) {
```

Comment:
nit : invert clause to avoid nested ifs

Reply:
It will complicate code a little because we will need to return `new CloudPathGrpcDecorator(credentials);` in 2 places, inside and outside inverted `if`.